### PR TITLE
#136 UniversalDecompressor hard-codes configuration file location

### LINKED ISSUE #136: UniversalDecompressor hard-codes configuration file location
In the `UniversalDecompressor` class of the wikimachine module, the path to the decompressor XML configuration file is hard-coded.  This makes it impossible for users (other than those compiling from source) to specify a decompressor of their choice.  The class should be rewritten to take a user-specified file location.

### PR DESCRIPTION
**Pull request by mawiesne**
- removes the hardcoded string PROPERTIES_PATH from `UniversalDecompressor`
- adds a new constructor to `UniversalDecompressor` taking a Path reference for custom XML file configs
- adds new test class `UniversalDecompressorTest`, previously totally untested
- demonstrates the handling of custom `ar` archive approach via decompressor-ar.xml, as 7z is not present on many platforms by default
- adjusts GZipDecompressor / BZip2Decompressor to check both ways to read an archive: classpath or external (file) location
- improves JavaDoc along the path

**What's in the PR**
* resolves #136 

**How to test manually**
* `mvn clean test`

**Automatic testing**
* [x] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: dkpro-jwpl-wikimachine/src/main/java/org/dkpro/jwpl/wikimachine/decompression/BZip2Decompressor.java [hunk 1]
```diff
@@ -35,19 +37,17 @@ public class BZip2Decompressor
...
-         * http://lucene.apache.org/tika/xref/org/apache/tika/parser /pkg/bzip2
-         * /CBZip2InputStream.html
-         */
-        inputStream.skip(2);
-        outputStream = new BZip2CompressorInputStream(inputStream);
-
-        return outputStream;
-
+        final InputStream in;
+        if (fileName.contains(File.separator)) {
```
Comment (rzo1) [inline dkpro-jwpl-wikimachine/src/main/java/org/dkpro/jwpl/wikimachine/decompression/BZip2Decompressor.java:41]: The `getResourceAsStream` doesn't explicitly state, what is considered as a valid `name` of a resource. If we look into `getResources` the [JavaDoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ClassLoader.html#getResource(java.lang.String)) states:

> The name of a resource is a '/'-separated path name that identifies the resource.

`File.separator` will return `/` on a *unix-oid system, so checking by `/` to decide if file system orr classloader won't work in all cases.

Maybe we can introduce an additional variable to indicate whether to check via classloader or handle it as a file-system resource?

Comment (reckart) [reply dkpro-jwpl-wikimachine/src/main/java/org/dkpro/jwpl/wikimachine/decompression/BZip2Decompressor.java:41]: How about using an URL?

**Review by rzo1 — APPROVED** [review]

**Comment by mawiesne** [comment]
Ping @logological

**Comment by logological** [comment]
@mawiesne Sorry, I'm no longer qualified to review this patch—I opened the original issue 8 years ago and my understanding of it (and the underlying code) has faded.

**Pull request merged**